### PR TITLE
Update program-derived-addresses.md

### PR DESCRIPTION
**Pull request by markdamasco**
Error Propagation
Actual account size to determine the rent-exempt amount.

### Summary of Changes

Fixes #

### CODE REVIEW

File: content/courses/native-onchain-development/program-derived-addresses.md [hunk 1]
```diff
@@ -104,25 +103,16 @@ function. If the output of `create_program_address` is not a valid PDA, then the
...
-    for _ in 0..std::u8::MAX {
-        {
-            let mut seeds_with_bump = seeds.to_vec();
-            seeds_with_bump.push(&bump_seed);
-            match Self::create_program_address(&seeds_with_bump, program_id) {
-                Ok(address) => return Some((address, bump_seed[0])),
-                Err(PubkeyError::InvalidSeeds) => (),
-                _ => break,
-            }
+    for bump in (0..=std::u8::MAX).rev() {
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/program-derived-addresses.md:106]: What does the `=` do here?

File: content/courses/native-onchain-development/program-derived-addresses.md [hunk 2]
```diff
@@ -631,16 +621,17 @@ let pda_counter = next_account_info(account_info_iter)?;
...
+const MAX_TITLE_LENGTH: usize = 256;
+const MAX_DESCRIPTION_LENGTH: usize = 1024;
 
-if MovieAccountState::get_account_size(title.clone(), description.clone()) > account_len {
-    msg!("Data length is larger than 1000 bytes");
-    return Err(ReviewError::InvalidDataLength.into());
+impl MovieAccountState {
+    pub fn get_account_size(title: &str, description: &str) -> usize {
+        8 + 1 + (4 + title.len().min(MAX_TITLE_LENGTH)) + (4 + description.len().min(MAX_DESCRIPTION_LENGTH))
+    }
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/program-derived-addresses.md:634]: We should use INIT_SPACE for this. https://www.anchor-lang.com/docs/space (scroll down to 'The InitSpace macro
').

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Thanks for your submission! A few small changes needed.

File: content/courses/native-onchain-development/program-derived-addresses.md (hunk 1 above)
Comment (markdamasco) [reply content/courses/native-onchain-development/program-derived-addresses.md:106]: range inclusion of all possible u8-bit values ie not excludes the end point.

**Pull request closed without merging**